galaxy_importer/collection.py

```diff
@@ -261,10 +261,29 @@ def _build_docs_blob(self):
                 for f in doc_files
             ]
 
+        ex_env_path = os.path.join(self.path, 'meta', 'execution_environment.yml')
+        execution_environment = {}
+        if not os.path.exists(ex_env_path):
+            self.log.info('No execution environment found.')
+        else:
+            try:
+                self.log.info('Linting execution environment data')
+                linting_result = markup_utils.lint_execution_environment_file(ex_env_path)
+                for line in linting_result:
+                    self.log.warning(line)
+            except Exception as e:
+                self.log.error(f'Error during linting of execution environment metadata: {e}')
+
+            try:
+                execution_environment = markup_utils.get_execution_environment(ex_env_path)
+            except Exception as e:
+                self.log.error(f'Error during parsing of execution environment metadata: {e}')
```

Comment:
I recommend pulling this out into something like `CollectionLoader._process_execution_environment()` that returns `execution_environment`. It will also make it easier to unit test without all the extra mocks.

Reply:
Thanks!